### DISCUSSION TITLE
More document properties

### DISCUSSION THREAD

**Discussion by arjunpaudyal** [sumatrapdfreader/sumatrapdf#2671] [Ideas]
More document properties CTRL + D or File >> Properties

Current document information is very few or limited. I would like to request to add feature to extend the document properties as it is displayed in Adobe Reader. 

More properties on :

* Fonts => Embedded, and used but not embedded (Fonts tab in Adobe Reader, but just name list. Adobe's is quite slow_
* Author
* Encryption -> if encrypted with User password /Master password (Security Tab in Adobe Reader)

* Wrap long filename, and description
* Make the property sizable (when its too wide due to filename, or description, or creator name)

**Comment by GitHubRulesOK** [comment]
Fonts should be similar to Adobe
Author is usually shown if embedded
![image](https://user-images.githubusercontent.com/12754913/172079081-320c761b-caea-4a91-be8f-4499517a15f3.png)

**Reply by arjunpaudyal** [reply]
![Sumatra3](https://user-images.githubusercontent.com/4165417/172097136-8be8c709-8c7f-44b3-9201-6c1cbf3b6bb1.JPG)
![AdobeReader](https://user-images.githubusercontent.com/4165417/172097162-36432633-7d41-4336-8ce8-5cb912213fdb.JPG)

That is not true in my case. I have attached SumatraPDF vs Adobe Reader file properties screen shot.

My current version is 3.3.3. I will try out with the newest available one.

**Comment by arjunpaudyal** [comment]
Font Information Display is available on 3.4.5.

Other request remains intact : 
* Encryption information availability
* Wrap too long file names or creator in File Properties window

**Reply by GitHubRulesOK** [reply]
Encryption is an odd one it is encrypted when not open
or it is decrypted when viewing otherwise its not viewable
so if you needed a password to open then it will be re-encrypted by password after changes
basically  adding (secured) to the title is only needed when using secure acrobat, SumatraPDF should generally be secure :-) see https://github.com/sumatrapdfreader/sumatrapdf/issues/1606#

**Comment by dominiosc** [comment]
Please consider to show keywords tag and be able to select the text to copy instead of all, or at least by individual property.

https://www.exiftool.org/TagNames/PDF.html

This pdf use keyword tag
https://apastyle.apa.org/instructional-aids/abstract-keywords-guide.pdf

**Reply by kjk** [reply]
This is tracked by https://github.com/sumatrapdfreader/sumatrapdf/issues/1394 and you can vote for it at https://sumatrapdf.canny.io/feature-requests/p/show-more-document-properties

**Reply by GitHubRulesOK** [reply]
The example above of

Keywords: APA Style; 7th edition; abstract; keywords

![image](https://user-images.githubusercontent.com/12754913/173207594-74b44a9e-cdcd-4872-a4e7-d1f7ebaec9bb.png)

 seems not much value when the file [abstract-keywords-guide.pdf](https://apastyle.apa.org/instructional-aids/abstract-keywords-guide.pdf) is already open and we can see its 7th edition

PDF Keywords were primarily a field for Doc Management Indexing. Certainly should not be a paragraph

**Reply by dominiosc** [reply]
That file is just a simplified example; there are files with many keywords.
If it is not displayed in the properties, we need to open the file in other applications, which is inconvenient.

**Reply by GitHubRulesOK** [reply]
Idoesnot have to be inconvenient just send the filename to exiftool like here 
<img width="1552" height="579" alt="image" src="https://github.com/user-attachments/assets/a3f58395-f63a-4acc-a903-e986e7670a27" />

So in latest pre-relese I can just allocate x to run exiftool and you can define with its command lines exactly what you wish to be shown.
```
ExternalViewers [
	[
		CommandLine = C:\Users\WDAGUtilityAccount\Desktop\Apps\Programming\c#\z80\plus\exiftool\exiftool.exe -k "%1"
		Name = e&Xiftool
		Filter = *.*
                Key = x
	]
]
```

**Comment by meteorquake** [comment]
There is one addition, which is the active page number. I use Ctrl-D to get a draft citation reference when I copy and quote from some PDF, and the active page is essential to have with this. If the PDF distinguishes between the printed page number and the PDF page number as some do, then both are desirable, so long as they are clear.
Also it would be useful if the escape key could close the dialog so you can do Ctrl-C then hit escape (you might want to have a hover tip over the copy to clipboard button saying that Ctrl-C will also copy).
Cheers, Dav,d

**Comment by l3w** [comment]
Hi to everyone,
I am very happy to use SUMATRA; light and fast.
I'm a modest developer (I write CLIPPER sources using Qt framework and create PDF with libHaru) so I haven't great knowledge about GUI used, so sorry me if I'm wrong.

About "pdf properties" window (I read SumatraProperties.cpp, CreatePropertiesWindow), why not use a window with VHscrollbars?
In this way SUMATRA can add/show a lot of properties without worry about window size (so, computing info positioning) or
using many Tab window ("General", "Fonts", "Security", "Other", etc.)
With only one button (Copy to clipboard) all informations can be exported, like current behaviour.
I believe that this solution follows the spirit of SUMATRA: essential, light and fast.

Many thanks for your job.
Best regards
Luigi
